src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Services/AzureSqlVMAdapter.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            var resp = Communicator.List();
+            return resp.Select((s) =>
+            {
+                return CreateSqlVirtualMachineModelFromResponse(s);
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Gets a list of all the sql virtual machines in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <returns>A list of all the sql virtual machines</returns>
+        public List<AzureSqlVMModel> ListSqlVirtualMachineByResourceGroup(string resourceGroupName)
+        {
+            var resp = Communicator.ListByResourceGroup(resourceGroupName);
+            return resp.Select((s) =>
+            {
+                return CreateSqlVirtualMachineModelFromResponse(s);
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Gets a sql virtual machine in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <param name="serverName">The name of the sql virtual machine</param>
+        /// <returns>The sql virtual machine</returns>
+        public AzureSqlVMModel GetSqlVirtualMachine(string resourceGroupName, string sqlVirtualMachineName)
+        {
+            var resp = Communicator.Get(resourceGroupName, sqlVirtualMachineName);
+            return CreateSqlVirtualMachineModelFromResponse(resp);
+        }
+
+        /// <summary>
+        /// Convert a Management.SqlVirtualMachine.Models.SqlVirtualMachineModel to AzureSqlVirtualMachineModel
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group the sql virtual machine is in</param>
+        /// <param name="resp">The management client sql virtual machine response to convert</param>
+        /// <returns>The converted sql virtual machine model</returns>
+        private static AzureSqlVMModel CreateSqlVirtualMachineModelFromResponse(SqlVirtualMachineModel resp)
+        {
+            // Extract the resource group name from the ID.
+            // ID is in the form:
+            // /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rgName/providers/Microsoft.SqlVirtualMachine/SqlVirtualMachine/sqlVirtualMachineName
+            string[] segments = resp.Id.Split('/');
+
+            AzureSqlVMModel model = new AzureSqlVMModel(segments[4])
+            {
+                Name = resp.Name,
+                Location = resp.Location,
+                Sku = resp.SqlImageSku,
+                Offer = resp.SqlImageOffer,
+                VirtualMachineId = resp.VirtualMachineResourceId,
+                SqlManagementType = resp.SqlManagement,
+                LicenseType = resp.SqlServerLicenseType,
+                Tags = TagsConversionHelper.CreateTagDictionary(TagsConversionHelper.CreateTagHashtable(resp.Tags), false),
+                ResourceId = resp.Id,
+                WsfcDomainCredentials = resp.WsfcDomainCredentials
```

Comment:
>WsfcDomainCredentials = resp.WsfcDomainCredentials [](start = 16, length = 50)

Why would we return credentials back from response? This should be empty all teh time, I believe.